src/content/guides/shimming.md

```diff
@@ -13,233 +13,386 @@ related:
   }
-  // ... rest of your webpack config
-};
+
+  document.body.appendChild(component());
+```
+
+__project__
+
+``` diff
+  webpack-demo
+  |- package.json
+  |- webpack.config.js
+  |- /dist
+  |- /src
+    |- index.js
+    |- globals.js
++   |- polyfills.js
++   |- fetch.js
+  |- /node_modules
 ```
 
-An alternative is to create a new entry file and manually import these packages.
+__src/polyfills.js__
 
 ```javascript
-// src/polyfills.js
 import 'babel-polyfill';
 import 'whatwg-fetch';
 ```
 
-```javascript
-// webpack.config.js
-module.exports = {
-  entry: {
-    polyfills: './src/polyfills.js',
-    main: './src/index.js'
-  }
-  // rest of your webpack config
-};
-```
+With that in place, we can add the logic to conditionally load `src/polyfills.js` depending on whether or not. How you make this decision depends on the technologies and browsers you need to support. We'll just do some simple testing before we run our fetching script:
 
-In your html file you need to conditionally load the `polyfills.js` file before your bundle. How you make this decision depends on the technologies and browsers you need to support.
+__src/index.js__
 
-```html
-<script>
-  var modernBrowser = (
-    'fetch' in window &&
-    'assign' in Object
-  );
+``` diff
+  function component() {
+    var element = document.createElement('div');
 
-  var scripts = [ '/main.js' ];
+    element.innerHTML = join(['Hello', 'webpack'], ' ');
 
-  if (!modernBrowser) {
-    scripts.unshift('/polyfills.js');
+    return element;
   }
 
-  scripts.map(function(src) {
-    var scriptElement = document.createElement('script');
-    scriptElement.async = false;
-    scriptElement.src = src;
-    document.head.appendChild(scriptElement);
-  });
-</script>
+  document.body.appendChild(component());
++
++ let modernBrowser = (
++   'fetch' in window &&
++   'assign' in Object
++ );
++
++ if ( !modernBrowser ) {
++   import('./polyfills.js', /* webpackChunkName: "polyfills" */).then(
++     import('./fetch.js' /* webpackChunkName: "fetch" */)
++   )
++
++ } else import('./fetch.js' /* webpackChunkName: "fetch" */)
+```
+
+__src/fetch.js__
+
+``` js
+fetch('https://jsonplaceholder.typicode.com/users')
+  .then(response => response.json())
+  .then(json => {
+    console.log('We retrieved some data! AND we\'re confident it will work on a variety of browser distributions.')
+    console.log(json)
+  })
+  .catch(error => console.error('Something went wrong when fetching this data: ', error))
 ```
 
-T> Any script added dynamically like in the example above will run as soon as it's parsed, but we need our polyfill to run before our bundle. This is why we are setting `async` to `false` for each script.
+Now, if we run our build a few more bundles will be emitted and everything should still run smoothly in the browser. Note that this set up could likely be improved upon but it should give you a good idea of how you can provide polyfills only to the users that need them.
 
```

Comment:
I really like the reorganization on this section, kudos!

The only confusing part is this one. First thing, I would put the content of `fetch.js` before. When coming across `index.js` I was confused because I didn't know what `fetch.js` did.

Other thing, I don't know how applicable the example is to real cases. I know that the script on index approach is not as _clean_ and diverge more form webpack documentation, but in general is the kind of loading strategy devs will need since usually the whole bundle relies on native APIs and not just small parts of it. Food for thought I guess.

Reply:
Yeah, @asapach mentioned the same thing -- I think I'll switch it back to the original example you had.